### ISSUE TITLE
Restore per-device options being reachable through HyprCtl.

### ISSUE THREAD

**Issue by Artifechs** [hyprwm/Hyprland#7080]
### Description

It seems we lost access to per-device config options since the new curly braces format was adopted.

So this does not work anymore:

    hyprctl getoption device:my-device:sensitivity
    hyprctl keyword device:my-device:sensitivity 1

Getting/setting these values on-demand is important for accessibility, at least mouse sensitivity.
    
As far as I can tell, the issue lies in [HyprCtl.cpp](https://github.com/hyprwm/Hyprland/blob/bc86afea7e177a7ad4335737417fc0e468698dc4/src/debug/HyprCtl.cpp#L1315):

    const auto VAR = g_pConfigManager->getHyprlangConfigValuePtr(curitem);
    
It only checks for "ConfigValue", but per-device options are stored as "SpecialConfigValue" in [ConfigManager.cpp](https://github.com/hyprwm/Hyprland/blob/bc86afea7e177a7ad4335737417fc0e468698dc4/src/config/ConfigManager.cpp#L567):

    m_pConfig->addSpecialConfigValue("device", "sensitivity", {0.F});
   
Any chance this lost functionality will be restored?

**Comment by vaxerski** [comment]
> hyprctl keyword device:my-device:sensitivity 1

the syntax has been changed a while back and is documented on the wiki:
```
hyprctl keyword 'device[my-device]:sensitivity 1'
```

for getoption, there's #5724

**Comment by Artifechs** [comment]
It's your project, I respect 100% if you just don't feel like solving this issue, but I have some more information. 

The current state of affairs seems to be that the functionality was lost due to refactoring, an attempt has been made to reimplement half of it, i.e. the `keyword` part, but it currently doesn't work, at least not for everyone, and is not documented in a very obvious place.

**Setting per-device values with hyprctl:**

The new syntax is not mentioned in the wiki on the [hyprctl page](https://wiki.hyprland.org/Configuring/Using-hyprctl/), nor on the [keywords page](https://wiki.hyprland.org/Configuring/Keywords/#per-device-input-configs) under device-specific configs. I still haven't been able to find it. But at least it's mentioned in this issue, so people can find it.

However, the command you mentioned doesn't work. I can set this in the .conf:

```
device {
    name = kensington-eagle-trackball 
    sensitivity = -0.5
    accel_profile = flat
}
```

But none of these commands do anything. I just get an `ok` and the sensitivity remains the same:
```
hyprctl keyword 'device[kensington-eagle-trackball]:sensitivity' -1
hyprctl keyword 'device[kensington-eagle-trackball]:sensitivity' 0
hyprctl keyword 'device[kensington-eagle-trackball]:sensitivity' 1
hyprctl keyword 'device[kensington-eagle-trackball]:sensitivity' 999
```

I can even put any garbage in there as the device name, and it just says `ok`:
```
hyprctl keyword 'device[otter-nose-pointer]:sensitivity' 1
```

**Getting per-device values with hyprctl:**

The solution for `getoption` you linked to is a thread of people talking about how they also are missing this functionality, and suggesting just programmatically changing the .conf file, which is circumventing `hyprctl`. 

**In conclusion:**

We could just change the .conf file in script for getting/setting these values and not bother with `hyprctl` altogether, I suppose that's an actual solution. Again, I respect if you're just not interested in making this work with `hyprctl`, but if that's the case, I'd highly recommend making it clear in the wiki that it's no longer supported, on purpose.

**Version info, if it matters:**

Hyprland, built from branch  at commit 918d8340afd652b011b937d29d5eea0be08467f5  (flake.lock: update).
Date: Tue Jun 25 12:06:02 2024
Tag: v0.41.2, commits: 4886

**Comment by vaxerski** [comment]
it doesnt work cuz I forgot you need the `-r` flag to hyprctl

**Comment by Artifechs** [comment]
True, this works:

```
hyprctl -r keyword 'device[kensington-eagle-trackball]:sensitivity' -1
```

That's definitely better than using `sed` to write changes to the `.conf` file. So all we need is the documentation available somewhere it can easily be found, as well as a reimplementation of `getoption`.

**Comment by Artifechs** [comment]
This doesn't work, though:

```
hyprctl -r keyword 'device[kensington-eagle-trackball]:sensitivity' -0.5
```

It seems it doesn't accept decimals.

**Comment by vaxerski** [comment]
> So all we need is the documentation available somewhere it can easily be found

MRs welcome

> It seems it doesn't accept decimals. If all this is mentioned in the wiki, I'm sorry, but I just can't find it anywhere.

`-` is parsed as an option marker, typical shell behavior:
```
hyprctl -r -- keyword 'device[kensington-eagle-trackball]:sensitivity' -0.5
```

**Comment by Artifechs** [comment]
Brilliant, thanks! 

PR here: https://github.com/hyprwm/hyprland-wiki/pull/728
